streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskExecutor.java

```diff
@@ -234,6 +244,15 @@ private void updateTaskCommitMetadata(final Map<TopicPartition, OffsetAndMetadat
         }
     }
 
+    private void commitSuccessfullyProcessedTasks() {
+        if (processingMode == AT_LEAST_ONCE && !tasks.successfullyProcessed().isEmpty()) {
```

Comment:
The check on the processingMode happens in two places, which seems redundant to me: e.g. if it is not ALOS, then the `successfullyProcessed()` should always be empty here. I think we can simply this line to only consider the second condition.